src/main/java/com/moodybluez/enterprise/dao/MoodDAO.java

```diff
@@ -0,0 +1,52 @@
+package com.moodybluez.enterprise.dao;
+
+import com.moodybluez.enterprise.dto.Mood;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+@Component
+public class MoodDAO implements IMoodDAO{
```

Comment:
Created new DAO class. Added logic from service class which should have been actually in DAO (this) layer.
Citation - "The Data Access Object (DAO) pattern is a structural pattern that allows us to isolate the application/business layer from the persistence layer" - [Baeldung](https://www.baeldung.com/java-dao-pattern)